SIPS/sip-futures_reference_price_methodology.md

```diff
@@ -0,0 +1,63 @@
+
+The forumla for the reference price is:
+\\[ 
+price = \begin{cases}
+\frac{d_{1} \ - \ X}{d_{1} \ - \ d_{0}} \cdot P_1 \ + \ \frac{d_{0} \ + \ X}{d_{1} \ - \ d_{0}} \cdot P_2 &   \ \mbox{if } \ \ \  X \leq d_1 \\
+\\
+\frac{d_{2} \ - \ X}{d_{2} \ - \ d_{1}} \cdot P_2 \ + \ \frac{X \ - \ d_{1}}{d_{2} \ - \ d_{1}} \cdot P_3 & \ \mbox{if } \ \ \ 0 \lt d_1 \lt X \\
+
+\end{cases}
+\\]
+With the inputs below:
+X: # of days prior to expiration to achieve zero weight in in the expiring contract (default of 5 days)
+DTE0: Days since the prior month contract expired
+DTE1: Days remaining for the current front month contact
+DTE2: Days remaining for the current 2nd month contract
+P0: Orderbook mid-price of the current front month contract
+P1: Orderbook mid-price of the current 2nd month contract
+p2: Orderbook mid-price of the current 3rd month contract
+
+There implications for fee reclamation with this approach, given that futures markets close, these Synths will use the next price fee reclamation mechanism. This requires that during market closures the Chainlink aggregator contract published a stable price. Due to the continuous time aspect of the calculation the reference price will continually update even durng market closures. This requires the node operators to subscribe to a market closure data feed to ensure the published price on-chain does not deviate outside a pre-defined range (likely 5bps) during market closures, which would trigger a next price update and cause all orders to fill at a stale price. There is a further implication for circuit breakers and other out of cycle market closures as these events would not be covered by the market closure data feed, in the case of a circuit breaker or other unscheduled market closure the data providers would continue to publish stale prices requiring these markets to be closed manually via the protocolDAO.
```

Comment:
Would a potential list of the circuit breaker situations / other out of cycle market closures be recorded within the SIP here for reference?

Reply:
We could provide some examples, but I don't think they will ever be exhaustive.